cristin-organization/src/main/java/no/unit/nva/cristin/organization/FetchCristinOrganizationHandler.java

```diff
@@ -2,54 +2,73 @@
 import nva.commons.apigateway.RequestInfo;
 import nva.commons.apigateway.exceptions.ApiGatewayException;
 import nva.commons.apigateway.exceptions.BadRequestException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.core.Environment;
 import nva.commons.core.JacocoGenerated;
 import nva.commons.core.paths.UriWrapper;
 
 import java.net.HttpURLConnection;
 import java.net.URI;
-import java.util.Optional;
 import java.util.regex.Pattern;
 
 import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_BACKEND_FAILED_WITH_STATUSCODE;
 import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID_FOUR_NUMBERS;
 import static no.unit.nva.cristin.model.Constants.BASE_PATH;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_API_BASE;
 import static no.unit.nva.cristin.model.Constants.DOMAIN_NAME;
 import static no.unit.nva.cristin.model.Constants.HTTPS;
+import static no.unit.nva.cristin.model.Constants.NOT_FOUND_MESSAGE_TEMPLATE;
 import static no.unit.nva.cristin.model.Constants.ORGANIZATION_PATH;
+import static no.unit.nva.cristin.model.Constants.UNITS_PATH;
 import static no.unit.nva.cristin.model.JsonPropertyNames.IDENTIFIER;
+import static no.unit.nva.model.Organization.ORGANIZATION_CONTEXT;
+import static no.unit.nva.model.Organization.ORGANIZATION_IDENTIFIER_PATTERN;
 
-public class FetchCristinOrganizationHandler extends ApiGatewayHandler<Void, Organization> {
+public class FetchCristinOrganizationHandler extends CristinQueryHandler<Void, Organization> {
 
-    public static final String IDENTIFIER_PATTERN = "^(?:[0-9]{1,5}\\.){3}[0-9]{1,3}$";
-    public static final Pattern PATTERN = Pattern.compile(IDENTIFIER_PATTERN);
+
+    public static final Pattern PATTERN = Pattern.compile(ORGANIZATION_IDENTIFIER_PATTERN);
     private final transient CristinApiClient cristinApiClient;
 
     @JacocoGenerated
     public FetchCristinOrganizationHandler() {
-        this(new CristinApiClient());
+        this(new CristinApiClient(), new Environment());
     }
 
-    public FetchCristinOrganizationHandler(CristinApiClient cristinApiClient) {
-        super(Void.class);
+    public FetchCristinOrganizationHandler(CristinApiClient cristinApiClient, Environment environment) {
+        super(Void.class, environment);
         this.cristinApiClient = cristinApiClient;
     }
 
+    @SuppressWarnings("PMD.AvoidThrowingNewInstanceOfSameException")
     @Override
     protected Organization processInput(Void input, RequestInfo requestInfo, Context context)
             throws ApiGatewayException {
-        Organization result;
         validateThatSuppliedParamsIsSupported(requestInfo);
+        final String identifier = getValidId(requestInfo);
         try {
-            result = getTransformedOrganizationFromCristin(getValidId(requestInfo));
+            Organization organization = cristinApiClient.getOrganization(new UriWrapper(HTTPS,
+                    CRISTIN_API_BASE)
```

Comment:
Can this be moved to `UriUtils` and be reused? Or should we keep it as is?

Reply:
I mean the building of the Cristin uri

Reply:
Added to UriUtils